mdb-realm/src/helpers.ts

```diff
@@ -1,126 +1,36 @@
-    }
+function generateClusterName(): string {
+    return createHash("md5")
+        .update(`${process.env.GITHUB_RUN_ID}-${process.env.GITHUB_RUN_ATTEMPT}`)
+        .digest("base64")
+        .replace(/\+/g, "")
+        .replace(/\//g, "")
+        .toLowerCase()
+        .substring(0, 8);
 }
 
-async function execAtlasRequest(
-    atlasUrl: string,
-    method: urllib.HttpMethod,
-    route: string,
-    config: EnvironmentConfig,
-    payload?: any,
-): Promise<any> {
-    const url = `${atlasUrl}/api/atlas/v1.0/groups/${config.projectId}/${route}`;
-
-    const request: urllib.RequestOptions = {
-        digestAuth: `${config.apiKey}:${config.privateApiKey}`,
-        method,
-        headers: {
-            "content-type": "application/json",
-            accept: "application/json",
-        },
-    };
-
-    if (payload) {
-        request.data = JSON.stringify(payload);
-    }
-
-    const response = await urllib.request(url, request);
-
-    if (response.status < 200 || response.status > 300) {
-        throw new Error(`Failed to execute ${request.method} ${route}: ${response.status}: ${response.data}`);
-    }
-
-    return JSON.parse(response.data);
-}
-
-function getSuffix(requireDifferentiator = true): string {
-    const differentiator = core.getInput("differentiator", { required: requireDifferentiator });
-    if (differentiator) {
-        return createHash("md5")
-            .update(`${getRunId()}-${differentiator}`)
-            .digest("base64")
-            .replace(/\+/g, "")
-            .replace(/\//g, "")
-            .toLowerCase()
-            .substring(0, 8);
-    }
-
-    return "no-cluster";
-}
-
-function getRunId(): string {
-    return process.env.GITHUB_RUN_ID || "";
-}
-
-export function getConfig(requireDifferentiator = true): EnvironmentConfig {
+export function getConfig(): EnvironmentConfig {
     return {
         projectId: core.getInput("projectId", { required: true }),
         apiKey: core.getInput("apiKey", { required: true }),
         privateApiKey: core.getInput("privateApiKey", { required: true }),
         realmUrl: core.getInput("realmUrl", { required: false }) || "https://realm-dev.mongodb.com",
         atlasUrl: core.getInput("atlasUrl", { required: false }) || "https://cloud-dev.mongodb.com",
-        clusterName:
-            core.getInput("clusterName", { required: false }) ||
-            getSuffix(
-                /*requireDifferentiator*/
-                requireDifferentiator || (core.getInput("clusterName", { required: false }) ?? "").trim() === "",
-            ),
-        useExistingCluster: core.getInput("useExistingCluster", { required: false }).toLowerCase() === "true" || false,
+        clusterName: core.getInput("clusterName", { required: false }) || generateClusterName(),
     };
 }
 
 export async function createCluster(config: EnvironmentConfig): Promise<void> {
+    const clusters = await getClusters(config);
+    if (clusters.includes(config.clusterName)) {
```

Comment:
Why not equals here?

Reply:
`clusters` is an array of strings - we want to check if it contains the cluster name.

Reply:
Gosh. Okay. I don't know why I thought that was a string.